File: C#/Structures/ForwardList.cs
using System;

class ForwardList{
    public class Node{
        public int data;
        public Node next = null;
        public Node(int data){
            this.data  = data;
        }
    }

    private Node front = null;
    private Node back = null;
    int m_size;

    public ForwardList(){
        this.m_size = 0;
    }

    public void add(int data){
        Node node  = new Node(data);
        if(is_empty()){
            this.front = node;
            this.back = node;
        }else{
            this.back.next = node;
            this.back = node;
        }

        ++ this.m_size;
    }

    public void add_front(int data){
        Node node  = new Node(data);
        if(is_empty()){
            this.front = node;
            this.back = node;
        }else{
            node.next = this.front;
            this.front = node;            
        }
        ++this.m_size;
    }

    public void add_back(int data){
        Node node = new Node(data);
        if(is_empty()){
            this.front = node;
            this.back = node;
        }else{
            this.back.next = node;
            this.back = node;
        }

        ++ this.m_size;
    }

    public void remove(int data){
        try{
            if(is_empty()) throw System.Exception;

            if(data == this.front.data) remove_front();

            Node ptr = this.front;
            Node prev = ptr;
            
            while(ptr.next != null){
                if(ptr.data == data){
                    if(ptr.data == this.back.data) {
                        remove_back(prev);
                        return;
                    }

                    prev.next = ptr.next;
                    return;
                }
                prev = ptr;
                ptr = ptr.next;
            }
            -- this.m_size;

        }catch (System.Exception){
            throw; // handle exception
        }
    }

    private void remove_front(){
        if is_empty() return; // improve with exceptions
        Node temp = this.front;
        this.front = this.front.next; 
        temp = null;
        -- this.m_size;
    }

    private void remove_back(Node prev){
        if is_empty() return; // improve with exceptions
        Node temp = this.back;
        this.back = prev;
        temp = null;

        -- this.m_size;
    }
    
    public void display(){
        Node ptr = this.front;
        while(ptr.next != null){
            Console.Write(ptr.data + " ");
            ptr = ptr.next;
        }
    }
    
    bool is_empty(){return this.front == null;}

    public int size(){return this.m_size;}
}

class MainList{
    public static void Main(string [] args){
        ForwardList list = new ForwardList();

        for(int i = 0; i<10; ++i)
            list.add(i);
            
        list.remove(2);
        list.display();
    }
}

File: Swift/Structures/ForwardList.swift
class Node{
    var data:Int = 0;
    var next:Node? = nil;

    init(m_data:Int) {data = m_data;}
}

class ForwardList{
    private var front:Node? = nil;
    private var back:Node? = nil;
    private var m_size:Int = 0;

    func add(data:Int){
        var node:Node = Node(data);
        if isEmpty(){
            self.front = node;
            self.back = node;
        }
        else{
            self.back?.next = node;
            self.back = node;
        }
        self.m_size += 1;
    }

    func add_back(data:Int){
        let node:Node = Node(m_data:data);
        if isEmpty(){
            self.front = node;
            self.back = node;
        }
        else{
            self.back?.next = node;
            self.back = node;
        }
        self.m_size += 1;
    }

    func add_front(data:Int){
        var node:Node = Node(data);
        if isEmpty(){
            self.front = node;
            self.back = node;
        }
        else{
            node?.next = self.front;
            self.front = node;
        }
        self.m_size += 1;
    }

    func remove_front(){ // how GC is managed in swift?
        if isEmpty(){
            return;
        }
        self.front = self.front!.next;
        self.m_size -= 1;
    }

    // func remove_back(){
    //     if isEmpty(){
    //         return;
    //     }
    //     self.back = self.back.prev;
    //     self.m_size -= 1;
    // }

    func remove(data:Int){
        if isEmpty(){
            return;
        }
        else{
            var ptr:Node? = self.front;
            var prev:Node? = ptr;

            while (ptr?.next != nil){
                if(ptr!.data == data){
                    prev?.next = ptr?.next;

                    self.m_size -= 1;
                    return; 
                }
                prev = ptr;
                ptr = ptr?.next;
            }
        }
    }

    func display(){
        var ptr:Node? = self.front;
        while(ptr?.next != nil){
            print(" \(ptr!.data)");
            ptr = ptr?.next;
        }
    }

    func isEmpty()->Bool{return self.front == nil;}
}

var l = ForwardList();

var i:Int = 10;

while (i > 0){
    l.add_back(data:i);
    i -= 1;
}

l.display();